x-pack/test/cloud_security_posture_functional/agentless/create_agent.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CLOUD_CREDENTIALS_PACKAGE_VERSION } from '@kbn/cloud-security-posture-plugin/common/constants';
+import * as http from 'http';
+import expect from '@kbn/expect';
+import type { FtrProviderContext } from '../ftr_provider_context';
+import { setupMockServer } from './mock_agentless_api';
+// eslint-disable-next-line import/no-default-export
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const mockAgentlessApiService = setupMockServer();
+  const pageObjects = getPageObjects([
+    'common',
+    'cspSecurity',
+    'security',
+    'header',
+    'cisAddIntegration',
+  ]);
+
+  const CIS_AWS_OPTION_TEST_ID = 'cisAwsTestId';
+
+  const AWS_SINGLE_ACCOUNT_TEST_ID = 'awsSingleTestId';
+
+  describe('Agentless cloud', function () {
+    let cisIntegration: typeof pageObjects.cisAddIntegration;
+    let mockApiServer: http.Server;
+
+    before(async () => {
+      cisIntegration = pageObjects.cisAddIntegration;
+      mockApiServer = await mockAgentlessApiService.listen(8089); // Start the usage api mock server on port 8081
+    });
+
+    after(async () => {
+      await await pageObjects.cspSecurity.logout();
+      mockApiServer.close();
+    });
+
+    it(`should create agentless-agent`, async () => {
+      const integrationPolicyName = `cloud_security_posture-${new Date().toISOString()}`;
+      await cisIntegration.navigateToAddIntegrationCspmWithVersionPage(
+        CLOUD_CREDENTIALS_PACKAGE_VERSION
+      );
+
+      await cisIntegration.clickOptionButton(CIS_AWS_OPTION_TEST_ID);
+      await cisIntegration.clickOptionButton(AWS_SINGLE_ACCOUNT_TEST_ID);
+
+      await cisIntegration.inputIntegrationName(integrationPolicyName);
+
+      await cisIntegration.selectSetupTechnology('agentless');
+      await cisIntegration.selectAwsCredentials('direct');
+
+      await pageObjects.header.waitUntilLoadingHasFinished();
+
+      await cisIntegration.clickSaveButton();
+      await pageObjects.header.waitUntilLoadingHasFinished();
+
+      expect(
+        (await cisIntegration.cisAws.getPostInstallCloudFormationModal()) ? true : false
+      ).to.be(true);
+    });
+
+    it(`should create default agent-based agent`, async () => {
+      const integrationPolicyName = `cloud_security_posture-${new Date().toISOString()}`;
+
+      await cisIntegration.navigateToAddIntegrationCspmWithVersionPage(
+        CLOUD_CREDENTIALS_PACKAGE_VERSION
+      );
+
+      await cisIntegration.clickOptionButton(CIS_AWS_OPTION_TEST_ID);
+      await cisIntegration.clickOptionButton(AWS_SINGLE_ACCOUNT_TEST_ID);
+
+      await cisIntegration.inputIntegrationName(integrationPolicyName);
+
+      await cisIntegration.clickSaveButton();
```

Comment:
I believe Ricky already have test cases for [Agent-based agents.](https://github.com/elastic/kibana/blob/35872af38076d06a292c524d1467d296876bf706/x-pack/test/cloud_security_posture_functional/pages/cis_integrations/cspm/cis_integration_aws.ts#L103) Do we need this test case?

Reply:
The test you are referring to is not running in the agentless context.  This agent-based test is for creating an agent-based policy when agentless is available.  I would like us to have something to cover us for this scenario so we don't have another incident like the bug we introduced in serverless.